File: RosService/src/package/src/service_client.cpp
#include "ros/ros.h"
#include "package/count_magnitude.h"
#include <cstdlib>

int main(int argc, char **argv) {
    // Inisialisasi node
    ros::init(argc, argv, "service_client");

    // Inisialisasi handler node
    ros::NodeHandle nh;

    // Membuat client untuk memanggil service
    ros::ServiceClient client = nh.serviceClient<package::count_magnitude>("count_magnitude");

    // Membuat objek pesan untuk request
    package::count_magnitude srv;

    // input user
    std::cout << "input komponen X: ";
    std::cin >> srv.request.komponenx;

    std::cout << "input komponen Y: ";
    std::cin >> srv.request.komponeny;

    std::cout << "input komponen Z: ";
    std::cin >> srv.request.komponenz;

    
    if (client.call(srv)) {
        ROS_INFO("Magnitude : %f", srv.response.magnitude);
    } else {
        ROS_ERROR("Service call failed");
        return 1;
    }

    return 0;
}

File: RosService/  src/package/src/service_client.py
#!/usr/bin/env python

import rospy
from package.srv import count_magnitude

def service_client(komponenx, komponeny, komponenz):
    # Inisialisasi ros node
    rospy.init_node('service_client')
    rospy.wait_for_service('count_magnitude')

    try:
    
        count_magnitude_service = rospy.ServiceProxy('count_magnitude', count_magnitude)

        # service request
        request = count_magnitude._request_class()
        request.komponenx = komponenx
        request.komponeny = komponeny
        request.komponenz = komponenz

        # service call
        response = count_magnitude_service(request)

        # result display
        rospy.loginfo("Magnitude : %f" % response.magnitude)

    except rospy.ServiceException as e:
        rospy.logerr("Service call failed %s" % e)

if __name__ == "__main__":
    # input user
    komponenx = int(input("input X: "))
    komponeny = int(input("input Y: "))
    komponenz = int(input("input Z: "))


    service_client(komponenx, komponeny, komponenz)
